ui/app/rif/components/chainAddresses.js

```diff
@@ -65,6 +52,49 @@ class ChainAddresses extends Component {
     };
   }
 
+  componentDidMount () {
+    this.props.getConfiguration()
+      .then(configuration => {
+        const resolvers = Object.assign([], GET_RESOLVERS(configuration));
+        this.setState({
+          resolvers,
+        });
+      });
+    this.loadResolver();
+    this.props.getResolver(this.props.domainName, this.props.subdomainName)
+      .then(resolver => {
+        if (!resolver.pending) {
+          this.setState({
+            selectedResolverAddress: resolver.address.toLowerCase(),
+          });
+        }
+      });
+  }
+
+  timeoutToLoadResolver = () => setTimeout(async () => {
+    let resolver = await this.props.getResolver(this.props.domainName, this.props.subdomainName);
+    if (resolver.pending) {
+      this.timeouts.push(this.timeoutToLoadResolver());
```

Comment:
Just for clarification, is this a recursive function that breaks the recursion when resolver.pending is false?

Reply:
Yes! Or on the unmount (That kills all timeouts)